OneWelcomeExampleApp/Interactors/RegisterUserInteractor.swift

```diff
@@ -45,8 +45,19 @@ extension RegisterUserInteractor: RegisterUserInteractorProtocol {
         return userClient.identityProviders
     }
 
-    func startUserRegistration(identityProvider: IdentityProvider? = nil) {
-        userClient.registerUserWith(identityProvider: identityProvider, scopes: ["read", "openid"], delegate: self)
+    func startUserRegistration(identityProvider: IdentityProvider?) {
+        guard let identityProvider else {
+            let error = AppError(errorDescription: "No identity provider chosen.")
+            registerUserPresenter?.registerUserActionFailed(error)
+            return
+        }
+        
+        switch AllowedIdentityProviders(rawValue: identityProvider.identifier) {
+        case .stateless, .twoWayStateless:
+            userClient.registerStatelessUserWith(identityProvider: identityProvider, scopes: ["read", "openid"], delegate: self)
```

Comment:
can we have an enum for scopes values?

Reply:
no, as this is related to strings from the backend